jguerrara_lru_cache.py

```diff
@@ -0,0 +1,26 @@
+class LRUCache:
+
+    def __init__(self, capacity):
+        self.vals = dict()
+        self.max_size = capacity
+
+    def put(self, key, value):
+        if key in self.vals:
+            del self.vals[key]
+            self.vals[key] = value
+        else:
+            if len(self.vals) < self.max_size:
+                self.vals[key] = value
+            else:
+                del self.vals[list(self.vals)[0]]
```

Comment:
When eviction is required this copy operation changes the complexity to O(n) from O(1). It also requires creating a copy of all of the data and then immediately throws it away